git-hub

```diff
@@ -61,10 +61,31 @@ ERR   = 0
 # Output functions, all use the str.format() function for formatting
 ########################################################################
 
+def _encode(_string, encoding=None):
+    if not encoding:
+        encoding = sys.getfilesystemencoding()
```

Comment:
is there any reason why `encoding` is passed as a parameter? It seems it will be always assigned here

Reply:
If the caller specifies an encoding, it will be used. 
If they don't, then `encoding` starts as `None`, and `sys.getfilesystemencoding` will be used. 

I.e. I want to allow the caller to specify an encoding, but fall back to `sys.getfilesystemencoding` if no encoding was specified. (This preserves the `_decode` signature, but I don't know if we really need it. Will `sys.getfilesystemencoding` ever be wrong?)

(I also changed the name pattern from `__decode` to `_decode`, because the `__` triggers [name mangling](https://docs.python.org/3/reference/expressions.html?highlight=mangling#atom-identifiers) which limits where the methods can be called from, and I wanted to use the encode/decode methods for string/bytes conversions everywhere.)

Reply:
Fair enough, thanks for the explanation and clarification!